app/models/ability.rb

```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, Post, Public: true
+    can :read, Comment, Public: true
+
+    return unless user.present?
+
+    can :read, :all
+
+    return unless user.role == 'admin'
+
+    can :manage, :all
+  end
+end
```

Comment:
- [ ] Great work creating this Ability class. However, the implementation based on the requirements is not complete. Currently, a normal user who doesn't have admin rights can not delete a post that was created by the user and the same goes for a comment that was created by the user. Kindly rewrite the code in the ability class to factor in this requirement 👍 💻. 

```ruby
class Ability
  include CanCan::Ability

  def initialize(user)
    can :read, Post, Public: true
    can :read, Comment, Public: true

    return unless user.present?

    can :read, :all

    # Allow regular users to manage their own posts and comments
    can :manage, Post, author_id: user.id
    can :manage, Comment, author_id: user.id

    return unless user.role == 'admin'

    can :manage, :all
  end
end
```

Reply:
Thank you for suggesting the missing requirements. Surely I will fix it